TodoApp/src/TodoController.java

```diff
@@ -0,0 +1,64 @@
+import java.util.*;
+
+class TodoUpdateParams {
+    public String title;
+    public String startDate;
+    public String endDate;
+    public UUID id;
+}
+
+class TodoController{
+    public Todo createTodo(String title, String startDate, String endDate){
+        Todo newTodo = new Todo();
+
+        UUID uuid = UUID.randomUUID();
+
+        newTodo.setTitle(title);
+        newTodo.setStartDate(startDate);
+        newTodo.setEndDate(endDate);
+        newTodo.setId(uuid);
+
+        return newTodo;
+    }
+
+    public Todo updateTodo(TodoUpdateParams params, Todo todo){
+        if (params.title != null){
+            todo.setTitle(params.title);
+        }
+
+        if (params.endDate != null){
+            todo.setEndDate(params.startDate);
+        }
+
+        if (params.startDate != null){
+            todo.setStartDate(params.endDate);
+        }
+
+        return todo;
+    }
+
+    public Todo getTodoByIdFromList(List<Todo> todoList, String todoId){
+        Optional<Todo> foundItem = todoList.stream()
+                .filter(item -> Objects.equals(item.getId().toString(), todoId))
+                .findFirst();
+
+        return foundItem.orElse(null);
+    }
+
+    public void listTodosFromList(List<Todo> todoList){
+        for (int i = 0; i < todoList.toArray().length; i++) {
```

Comment:
Aqui você ao invés de utilizar o for normal voce pode utilizar o for in do Java

Exemplo:

```java
for(Todo t : todoList.toArray()){
       System.out.println(t);
}
```

Reply:
Pô, bem melhor